test/rubygems/test_gem_commands_install_command.rb

```diff
@@ -1004,4 +1004,13 @@ def test_handle_options_without
     assert_equal [:test, :development], @cmd.options[:without_groups]
   end
 
+  def test_handle_proxy_with_no_scheme
+    err = assert_raises OptionParser::InvalidArgument do
+      @cmd.handle_options %w[-p=foo.bar.com]
+    end
+
+    assert_match /invalid argument: -p=foo.bar.com/ , err.message
+    assert_match /Preface URLs with one of \["http", "https", "file", "s3"\]/ , err.message
```

Comment:
I've written what I think is a straightforward test, I just want to throw an exception and test that the message I get back contains the original message + the new bit. 

I was able to test my change in ```lib/rubygems/local_remote_options.rb``` manually, and it works. However, the second assertion (line 1013) doesn't evaluate to true. It's almost as if a mock/test double is being used, but if that's the case I can't find it.

If someone could lend me a helpful hand in getting this test to run correctly, I'd really appreciate it.

Reply:
@ajwann Have you tried printing both the exception that you've created as well as the string that you're comparing the assert with to see if there might be a typo?